node/bindnode/infer.go

```diff
@@ -186,10 +201,11 @@ func verifyCompatibility(seen map[seenEntry]bool, goType reflect.Type, schemaTyp
 
 		for i, schemaType := range schemaMembers {
 			goType := goType.Field(i).Type
-			if goType.Kind() != reflect.Ptr {
-				doPanic("union members must be pointers")
+			if ptr, nilable := ptrOrNilable(goType.Kind()); !nilable {
+				doPanic("union members must be nilable")
```

Comment:
I don't think this is the ideal error here - it's talking schema language about a Go struct, maybe that's the best we have? but istm that this would be better stated as "must be pointers or interfaces".

Not blocking, just noting that it's weird encountering this error in the test when it's refering to the Go type.